eslint-rules/src/rules/use-correct-component-overrides.ts

```diff
@@ -0,0 +1,136 @@
+import { Selectors } from '@angular-eslint/utils';
+import { ASTUtils, TSESLint, TSESTree } from '@typescript-eslint/experimental-utils';
+import * as fs from 'fs';
+import * as path from 'path';
+
+import { normalizePath } from '../helpers';
+
+export const useCorrectComponentOverridesRule: TSESLint.RuleModule<string, []> = {
+  meta: {
+    messages: {
+      shouldPointToBasicFile: 'Override should point to basic component file.',
+      testOverrideTemplateMissing:
+        'Did not find template override ({{ expectedTemplate }}) this spec is supposed to test.',
+      testOverrideTSMissing: 'Did not find TS import ({{ expectedTS }}) this spec is supposed to test.',
+      pointToBasicFile: 'Replace with basic component file.',
+    },
+    type: 'problem',
+    schema: [],
+    hasSuggestions: true,
+  },
+  create(context) {
+    const fileName = path.basename(normalizePath(context.getFilename()));
+    const fileBaseName = fileName.replace(/component.*/, 'component');
+    if (/\.component\.(?!(spec|stories))([a-z]+\.)?ts$/.test(fileName)) {
```

Comment:
Maybe you can make this regexp configurable via `eslintrc.json`? It doesn't really make sense to include `stories` here for the PWA, even though some projects might use it ;)

Reply:
Just realized that neither spec nor stories will define a Component with URL fields. But `spec` is needed for the if/else.